ocaml/xapi-guard/lib/disk_cache.ml

```diff
@@ -585,19 +567,29 @@ end = struct
     let now = Mtime_clock.now () in
     let root = cache_of typ in
     let* contents = get_all_contents root in
-    let* () =
-      contents
-      |> List.map (get_fs_action root now)
-      |> Lwt_list.iter_p (commit __FUNCTION__)
+    let actions, pending =
+      contents |> List.map (get_fs_action root now) |> List.split
     in
-    delete_empty_dirs ~delete_root:false root
+    let* () = Lwt_list.iter_p (commit __FUNCTION__) actions in
+    let* () = delete_empty_dirs ~delete_root:false root in
+    Lwt.return (List.filter_map Fun.id pending)
 end
 
 let setup typ read write =
-  let* () = Setup.retime_cache_contents typ in
-  let queue, push = Lwt_bounded_stream.create 2 in
+  let* pending = Setup.retime_cache_contents typ in
+  let capacity = 4096 in
```

Comment:
This looks like a rather large value for a list length.

Reply:
I've reduced this to 512, it should give less problems with the stack, while being big enough to handle most cases (512 VMs in a host with TPM that have written while restarting)